virtcontainers/qemu.go

```diff
@@ -1567,6 +1587,32 @@ func (q *qemu) resizeVCPUs(reqVCPUs uint32) (currentVCPUs uint32, newVCPUs uint3
 	return currentVCPUs, newVCPUs, nil
 }
 
+func (q *qemu) updateMemoryBalloon(sizeMB uint32) error {
+
+	if q.state.HasVFIODevice {
+		// When VFIO devices are used we dont want to get pages back to
+		// the host. Lets increase the balloon to a large amount of
+		// memory:
+		// - The guest memmory is not affected
+
+		// TODO: This is not required for qemu +v3.1.0 where "Inhibit
+		// ballooning" is implemented.
+
+		// Reference:
+		// https://www.ibm.com/support/knowledgecenter/en/SSZJY4_3.1.0/liabp/liabpmemoryballooning.htm
+		// https://bugs.launchpad.net/qemu/+bug/1762707
+		hostMB, err := q.hostMemMB()
+		if err != nil {
+			return err
+		}
+		sizeMB = uint32(hostMB)
+		q.Logger().WithField("memory-mb", sizeMB).Warnf("VM uses VFIO, balloon size will be total host memory")
```

Comment:
Again, I would like to see results of how things work when mdev devices are assigned. According to the bug that you referenced here, mdev device may or may not pin memory depending on the vendor driver. We should be testing all these scenarios as well.

Reply:
@amshinde sounds good! I'll contact you to define a set of test to verify.  Also will ping @awilliam
to review this logic and verify if is  better just move to qemu 3.1+

Reply:
In QEMU (3.1+) the balloon inhibitor is activated for any vfio device, whether it's a physical or mdev device.  While mdev devices are theoretically more compatible with ballooning, there is no guarantee that the vendor driver isn't sloppy at tracking pages actively used by the device in the guest, which could make them available to the balloon driver in the guest and result in the same sort of mismatches that we're trying to avoid.  It's safer to ignore that mdev device might be more balloon friendly for now, we'll need to develop new interfaces to expose if they ever truly are.

Reply:
@awilliam thank you, so I think the best a option is to move to qemu 3.1 first